modules/client/src/main/java/org/apache/ignite/internal/jdbc/JdbcConnection.java

```diff
@@ -0,0 +1,769 @@
+    private static final String SET_NETWORK_TIMEOUT_PERM = "setNetworkTimeout";
+
+    /** Statements modification mutex. */
+    private final Object stmtsMux = new Object();
+
+    /** Handler. */
+    private final JdbcQueryEventHandler handler;
+
+    /** Schema name. */
+    private String schema;
+
+    /** Closed flag. */
+    private volatile boolean closed;
+
+    /** Current transaction isolation. */
+    private int txIsolation;
+
+    /** Auto-commit flag. */
+    private boolean autoCommit;
+
+    /** Read-only flag. */
+    private boolean readOnly;
+
+    /** Current transaction holdability. */
+    private int holdability;
+
+    /** Connection properties. */
+    private final ConnectionProperties connProps;
+
+    /** Tracked statements to close on disconnect. */
+    private final Set<JdbcStatement> stmts = Collections.newSetFromMap(new IdentityHashMap<>());
+
+    /** Network timeout. */
+    private int netTimeout;
+
+    /** Retry limit. */
+    private final int retryLimit;
+
+    /** Reconnect throttling period. */
+    private final long reconnectThrottlingPeriod;
+
+    /** Reconnect throttling retries. */
+    private final int reconnectThrottlingRetries;
+
+    /** Query timeout. */
+    private final @Nullable Integer qryTimeout;
+
+    /** Ignite remote client. */
+    private final TcpIgniteClient client;
+
+    /**
+     * Constructor.
+     *
+     * @param handler Handler.
+     * @param props Properties.
+     */
+    public JdbcConnection(JdbcQueryEventHandler handler, ConnectionProperties props) {
+        this.connProps = props;
+        this.handler = handler;
+
+        autoCommit = true;
+
+        netTimeout = connProps.getConnectionTimeout();
+        qryTimeout = connProps.getQueryTimeout();
+        retryLimit = connProps.getRetryLimit();
+        reconnectThrottlingPeriod = connProps.getReconnectThrottlingPeriod();
+        reconnectThrottlingRetries = connProps.getReconnectThrottlingRetries();
+
+        holdability = HOLD_CURSORS_OVER_COMMIT;
+
+        schema = SchemaTable.DEFAULT_SCHEMA_NAME;
+
+        client = null;
+    }
+
+    /**
+     * Creates new connection.
+     *
+     * @param props Connection properties.
+     */
+    public JdbcConnection(ConnectionProperties props) {
+        this.connProps = props;
+        autoCommit = true;
+
+        String[] addrs = Arrays.stream(props.getAddresses()).map(HostAndPortRange::toString)
+            .toArray(String[]::new);
+
+        netTimeout = connProps.getConnectionTimeout();
+        qryTimeout = connProps.getQueryTimeout();
+        retryLimit = connProps.getRetryLimit();
+        reconnectThrottlingPeriod = connProps.getReconnectThrottlingPeriod();
+        reconnectThrottlingRetries = connProps.getReconnectThrottlingRetries();
+
+        client = ((TcpIgniteClient)IgniteClient
+            .builder()
+            .addresses(addrs)
+            .connectTimeout(netTimeout)
+            .retryLimit(retryLimit)
```

Comment:
retryLimit member field is used only here

Reply:
Removed them from class field list, tnx.